src/main/java/io/kestra/plugin/aws/kinesis/PutRecords.java

```diff
@@ -0,0 +1,295 @@
+            }
+        ),
+        @Example(
+            title = "Put multiple records as a JSON string to Amazon Kinesis.",
+            code = {
+                "streamName: \"mystream\"",
+                "records: |",
+                "   [",
+                "     {",
+                "       \\\"Data\\\": \\\"user sign-in event\\\",",
+                "       \\\"PartitionKey\\\": \\\"user1\\\"",
+                "     },",
+                "     {",
+                "       \\\"Data\\\": \\\"user sign-out event\\\",",
+                "       \\\"PartitionKey\\\": \\\"user1\\\"",
+                "     }",
+                "   ]"
+            }
+        ),
+        @Example(
+            title = "Put multiple records as an internal storage uri.",
+            code = {
+                "streamName: \"mystream\"",
+                "records: kestra://myfile.ion"
+            }
+        )
+    }
+)
+@Schema(
+    title = "Put multiple records to Amazon Kinesis."
+)
+public class PutRecords extends AbstractConnection implements RunnableTask<PutRecords.Output> {
+    private static final ObjectMapper MAPPER = JacksonMapper.ofIon()
+        .setSerializationInclusion(JsonInclude.Include.ALWAYS);
+
+    @PluginProperty(dynamic = false)
+    @NotNull
+    @Schema(
+        title = "Mark the task as failed when putting a record is unsuccessful.",
+        description = "If true, the task will fail when any record fails to be put."
+    )
+    @Builder.Default
+    private boolean failOnUnsuccessfulRecords = true;
+
+    @PluginProperty(dynamic = true)
+    @Schema(
+        title = "The name of the stream to add the records.",
+        description = "One of streamName of streamArn has to be set"
+    )
+    private String streamName;
+
+    @PluginProperty(dynamic = true)
+    @Schema(
+        title = "The ARN of the stream to add the records.",
+        description = "One of streamName of streamArn has to be set"
+    )
+    private String streamArn;
+
+    @PluginProperty(dynamic = true)
+    @Schema(
+        title = "List of record to put or internal storage uri to retrieve it.",
+        description = "A list of at least one record."
+    )
+    @NotNull
+    private Object records;
+
+    @Override
+    public Output run(RunContext runContext) throws Exception {
+        final long start = System.nanoTime();
+
+        List<Record> records = getRecordList(this.records, runContext);
+
+        PutRecordsResponse putRecordsResponse = putRecords(runContext, records);
+
+        // Set metrics
+        runContext.metric(Timer.of("duration", Duration.ofNanos(System.nanoTime() - start)));
+        runContext.metric(Counter.of("failedRecordCount", putRecordsResponse.failedRecordCount()));
+        runContext.metric(Counter.of("successfulRecordCount", records.size() - putRecordsResponse.failedRecordCount()));
+        runContext.metric(Counter.of("recordCount", records.size()));
+
+        // Fail if failOnUnsuccessfulRecords
```

Comment:
Fail before registering metrics, I don't think we will store them if the task fail.